### ISSUE TITLE
Project Cleanup

### ISSUE THREAD

**Issue by Fragilem17** [jonathanhart/oculus-ane#6]
Hi Guys,

I think we should clean out everything we don't need from this repository.

The demo I was making for example should only reflect the workings of our ANE and not of the separate joystick ANE. It should also just use the compiled ANE file and as little separate code as possible... Also it should use the implementation for each 3D engine's files that we put in the frameworks folder. (maybe making it a library project and from the demo's folder directly referencing the resulting swc)

I can rework the demo and provide something simpler and commit that.

But aside from the demo i think the OculusSDK should maybe not be completely mirrored here. What do you think? Shouldn't we just put the files we need from the SDK in our repository? the Tuscany demo for example can go i think.

I'll just list some of the stuff i would like to see, just comment on whether you all agree or not and then we will remove it.

<b>oculus_sdk\OculusSDK</b>
 &nbsp;&nbsp;remove the double folder maybe only keeping the OculusSDK 

<b>Win_OculusWorldDemo</b>
 &nbsp;&nbsp;remove the shortcut

<b>oculus_sdk\OculusSDK\Samples</b>
 &nbsp;&nbsp;remove Samples

<b>oculus_sdk\OculusSDK\Mac_OculusWorldDemo.app</b>
 &nbsp;&nbsp;remove Mac_OculusWorldDemo.app

remove if not needed, not sure: <b>3rdParty</b>

**Comment by BigPhilCombo** [comment]
I do agree that it could be cleaned up.  I think that anything related to
building the ane should be in the ane folder, so the AS3 api, the windows
project, the osx project, etc..  Then the ANE should be build directly into
a location that is used by the demo projects so that the file is not
duplicated in each project.  That way when the ANE gets updated the demos
get updated as well.  Also there are a few instances where absolute paths
are used, those should get changed to relative paths so everything works
right out of the box.

On Wed, Jun 26, 2013 at 11:43 AM, Tom Goethals notifications@github.comwrote:

> Hi Guys,
> 
> I think we should clean out everything we don't need from this repository.
> 
> The demo I was making for example should only reflect the workings of our
> ANE and not of the separate joystick ANE. It should also just use the
> compiled ANE file and as little separate code as possible... Also it should
> use the implementation for each 3D engine's files that we put in the
> frameworks folder. (maybe making it a library project and from the demo's
> folder directly referencing the resulting swc)
> 
> I can rework the demo and provide something simpler and commit that.
> 
> But aside from the demo i think the OculusSDK should maybe not be
> completely mirrored here. What do you think? Shouldn't we just put the
> files we need from the SDK in our repository? the Tuscany demo for example
> can go i think.
> 
> I'll just list some of the stuff i would like to see, just comment on
> whether you all agree or not and then we will remove it.
> 
> _oculus_sdk\OculusSDK_
>   remove the double folder maybe only keeping the OculusSDK
> 
> _Win_OculusWorldDemo_
>   remove the shortcut
> 
> _oculus_sdk\OculusSDK\Samples_
>   remove Samples
> 
> _oculus_sdk\OculusSDK\Mac_OculusWorldDemo.app_
>   remove Mac_OculusWorldDemo.app
> 
> remove if not needed, not sure: _3rdParty_
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/jonathanhart/oculus-ane/issues/6
> .

## 

Phillip Chertok

**Comment by Fragilem17** [comment]
Agreed! the build.bat for windows depends on the flex sdk being in flashDevelops program files path :-)
should we copy the latest flexSDK into the project? so everything can build from just checking out this repository?

ane
&nbsp;&nbsp;\- bin (contains the end result ANE and SWC file)
&nbsp;&nbsp;\- build (build scripts directly referencing the release folders, exporting to 'bin' above)
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;\-  tools (containing flex sdk, 7z.exe, xcopy.exe, ...)
&nbsp;&nbsp;\- mac
&nbsp;&nbsp;\- windows
&nbsp;&nbsp;\- shared
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;\-  OculusSDK
demos
&nbsp;&nbsp;\- away3d
&nbsp;&nbsp;\- minko
&nbsp;&nbsp;\- ...

The frameworks folder can maybe just be deleted, if we focus on 1 demo for each framework that shows all features but is kept to the bare minimum... i think that would be the easiest. at least to start.

**Comment by BigPhilCombo** [comment]
I don't think we should necessarily include the flex sdk or even the
OculusSDK for that matter, we should leave it up to the developer to obtain
those items themselves.  We could either allow for the user to modify some
configuration file that points to the appropriate location of those items,
or just indicate clearly where they need to be placed.

On Wed, Jun 26, 2013 at 12:16 PM, Tom Goethals notifications@github.comwrote:

> Agreed! the build.bat for windows depends on the flex sdk being in
> flashDevelops program files path :-)
> should we copy the latest flexSDK into the project? so everything can
> build from just checking out this repository?
> 
> ane
> - bin (contains the end result ANE and SWC file)
> - build (build scripts directly referencing the release folders,
>   exporting to 'bin' above)
>   - tools (containing flex sdk, 7z.exe, xcopy.exe, ...)
> - mac
> - windows
> - shared
>   - OculusSDK
>     demos
> - away3d
> - minko
> - ...
> 
> The frameworks folder can maybe just be deleted, if we focus on 1 demo for
> each framework that shows all features but is kept to the bare minimum... i
> think that would be the easiest. at least to start.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/jonathanhart/oculus-ane/issues/6#issuecomment-20060308
> .

## 

Phillip Chertok

**Comment by mrmmaclean** [comment]
Really we shouldn't have any libraries or folders or files that are not ours and not directly related to the plugin. There's licensing issues, and beyond that it just bloats our (very small) code library. People should be setting up their own projects to include necessary libraries, headers, etc. Even having the hard coded folder path to include the OVR.h in the Windows header as Fragilem17 had modified it was bad practice (and didn't even work with the way I had my libraries set up).

I use Flashbuilder 4.6 rather than Flashdevelop these days and I use Visual Studio for the C++ side of things, yet I learned how to create an ANE by using a C++ eclipse plugin for Flashbuilder. Keeping includes and imports as generic as possible is the best way for the code to work out-of-the-box for everyone's individual setups, also freeing us up from having to make all sorts of different types of project files (Flashdevelop, Flashbuilder, XCode, NetBeans, VS, etc, etc, etc.) Source code is source code, and if you can't use it effectively then it's time to either use the pre-built plugin provided or go back to learning how your IDE works with some Hello World tutorials.

There's also the fact that when we have various other SDKs (such as the Oculus SDK) in our repository we'll have to make sure we upload the newest versions when they do. That's not on us.

**Comment by BigPhilCombo** [comment]
Yes

Also for the demo projects, if external libraries are used I believe it is
better to rely on SWCs rather than actual source code.

On Wed, Jun 26, 2013 at 1:10 PM, mrmmaclean notifications@github.comwrote:

> Really we shouldn't have any libraries or folders or files that are not
> ours and not directly related to the plugin. There's licensing issues, and
> beyond that it just bloats our (very small) code library. People should be
> setting up their own projects to include necessary libraries, headers, etc.
> Even having the hard coded folder path to include the OVR.h in the Windows
> header as Fragilem17 had modified it was bad practice (and didn't even work
> with the way I had my libraries set up).
> 
> I use Flashbuilder 4.6 rather than Flashdevelop these days and I use
> Visual Studio for the C++ side of things, yet I learned how to create an
> ANE by using a C++ eclipse plugin for Flashbuilder. Keeping includes and
> imports as generic as possible is the best way for the code to work
> out-of-the-box for everyone's individual setups, also freeing us up from
> having to make all sorts of different types of project files (Flashdevelop,
> Flashbuilder, XCode, NetBeans, VS, etc, etc, etc.) Source code is source
> code, and if you can't use it effectively then it's time to either use the
> pre-built plugin provided or go back to learning how your IDE works with
> some Hello World tutorials.
> 
> There's also the fact that when we have various other SDKs (such as the
> Oculus SDK) in our repository we'll have to make sure we upload the newest
> versions when they do. That's not on us.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/jonathanhart/oculus-ane/issues/6#issuecomment-20064039
> .

## 

Phillip Chertok

**Comment by mrmmaclean** [comment]
Agreed, that way they will always work and the user doesn't need to compile any additional libraries.

**Comment by Fragilem17** [comment]
Okay, I'm totally hearing you guys. let's start removing then :-)

As I'm not the original starter of this repo (or what are they calling it
in git terms) I'd feel bad to just start pressing delete.

What would be a good way of agreeing on a ultimate folder structure?

Does one just create a fork or something deletes and adds some stuff and
when that is good we pull those changes to this repo?

Any other ways of effectively doing this? I am not used not being in the
same office as my fellow developers... Just shouting to each other is the
way I'd do it normally.

**Comment by mrmmaclean** [comment]
Just go for it! I am not a programmer by trade and this is literally the first code thing I've ever collaborated on so I'll leave it to the professionals :)

**Comment by BigPhilCombo** [comment]
Yes I suspect this is why Jonathan gave access to the repo so he did not
have to be burdened with all these details.  I wonder though if it makes
sense to create a new github account just for this.  It would have 2 repos
- one - oculus-ane - the other oculus-ane-examples?

On Wed, Jun 26, 2013 at 3:44 PM, mrmmaclean notifications@github.comwrote:

> Just go for it! I am not a programmer by trade and this is literally the
> first code thing I've ever collaborated on so I'll leave it to the
> professionals :)
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/jonathanhart/oculus-ane/issues/6#issuecomment-20074571
> .

## 

Phillip Chertok

**Comment by jonathanhart** [comment]
In my experience, repositories have started off as two github repos but
consolidated into one over time. So that's why I started out as a single
repository, which I think would still be the simplest provided the raw
source code necessary is not hidden inside the individual demos.

However, a better organization of the code to further reduce confusion
would be a good thing to address.

Also, I've been very swamped and out of town for a while celebrating my
anniversary with my gf so that's why I've been off the grid. I'll be happy
to let you guys do the reorganizing however you see fit.

**Comment by Fragilem17** [comment]
subversion and git do not play well together! aargh

**Comment by Fragilem17** [comment]
okay, cleaned it up a bit. Is anybody missing anything? more stuff that can be deleted?
the build.sh file will have to be modified to accommodate the new paths.

I haven't been testing things yet, and social life is calling.. hopefully tonight i can get some more time into this.
Next up is cleaning the demo folder so only 1 good simple demo that just shows of our library is in there. (removing all the FlashDevelop specific stuff among others)

**Comment by BigPhilCombo** [comment]
In regards to the Joystick use in the current demo,  might want to replace
that with the native GameInput API.  It would lock the demo to AIR 3.7 and
up but that way you can still use the joystick without relying on another
ANE.

On Thu, Jun 27, 2013 at 12:07 PM, Tom Goethals notifications@github.comwrote:

> okay, cleaned it up a bit. Is anybody missing anything? more stuff that
> can be deleted?
> the build.sh file will have to be modified to accommodate the new paths.
> 
> I haven't been testing things yet, and social life is calling.. hopefully
> tonight i can get some more time into this.
> Next up is cleaning the demo folder so only 1 good simple demo that just
> shows of our library is in there. (removing all the FlashDevelop specific
> stuff among others)
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/jonathanhart/oculus-ane/issues/6#issuecomment-20133667
> .

## 

Phillip Chertok

**Comment by Fragilem17** [comment]
But work only on android for the moment right?

**Comment by Fragilem17** [comment]
The old away3d demo is removed, a simple no-nonsense demo is added.
It features my take on away3D integration.

**Comment by Fragilem17** [comment]
It's that time for a big cleanup again with all the testing for DK2 we've been doing.
we need a common src folder for both Windows and OSX c++ code.
1 bin folder where the dll and .framework files are put.
all external libraries (Oculus SDK) should not be included in this repository.

Will start deleting and cleaning up. the osX project files will need to be refactored.

**Comment by Fragilem17** [comment]
taking inspiration from this mature ANE project that is both mac and windows compatible 
https://github.com/AS3NUI/airkinect-2-core

They only have one src folder for both the osx-xcode project and the visual studio to make use of.
